GKE Hub Policies

Sundi202 opened the pull request:
Policies implemented:
fleet_logging_default_mode_required → Ensures Fleet Observability logging mode is explicitly set to COPY or MOVE.
no_public_principals_binding - Prevents use of allUsers and allAuthenticatedUsers in IAM bindings.
no_public_principals_member -  Disallows assigning roles directly to public principals at the member level.
git_secure_auth - Enforces use of secure Git/OCI secret types and approved service account emails.
git_approved_HTTPS - Ensures Git/OCI traffic routes through an approved organizational HTTPS proxy.
git_approved_dir - Validates that policy_dir values align with organizationally approved paths.
pc_enable_required - Requires Policy Controller to be enabled.

Sundi202 committed:
GKE Hub Feature (Fleet Observability) -  require default logs routing mode COPY or MOVE (updated)

Sundi202 committed:
Security policies for google_gke_hub_feature

Sundi202 committed:
no_public_principals_binding policy for IAM binding

Sundi202 committed:
no_public_principals_member policy for IAM member

Sundi202 committed:
configmanagement security policies (git_secure_auth, git_approved_HTTPS, git_approved_dir, pc_enable_required)

Pat-Stuart commented on inputs/gcp/GKEHub/google_gke_hub_feature_membership/pc_enabled_required/c.tf:
You have used membership in your vars.rego as your "resource_value_name" therefore it needs to be in our standard format as this is what will be output when running your policies. We want policies to output a standard format so we want whatever attribute you have put in the vars.rego "resource_value_name" to be of the format "c" or "nc" with possible digits after if there are multiple resource instances.

This applies to all of the TF for this resource.

Sundi202 replied:
Addressed comment and fixed all the nc and c naming format.

Pat-Stuart commented on policies/gcp/GKEHub/google_gke_hub_feature_membership/git_approved_dir/policy.rego:
This is a configuration value and I would say is not security relevant

Sundi202 replied:
Removed the argument.

Pat-Stuart reviewed (changes requested):
Policies are largely fine however naming needs to be fixed up. There are 2 options here:

1. What I described in the comment, this will also apply to the resources that use "name" if you go that way.
2. You rebase or merge in the latest dev branch into this branch and use "name" for all. This will use the top level resource name so all should be correct

Sundi202 committed:
added policy_binding policy & authority_issuer policy

Sundi202 committed:
GKEHub: remove deprecated google_gke_hub_feature_membership/git_approved_dir

Sundi202 committed:
GKEHub: standardize resource_value_name to {c|nc}; fix authority_issuer; tighten RBAC/HTTPS controls

Sundi202 commented:
I have addressed all the comments raised in the previous review and previously added 4 policies.  I’ve also attached my documentation on the policies as a Word document, since I was running into some issues working with JSON formatting.
Please let me know if any further adjustments are needed. 

https://deakin365.sharepoint.com/:w:/r/sites/HardhatEnterprises2/Shared%20Documents/%F0%9F%95%B9%20Policy%20Deployment%20Engine/Documents/Service-Policy%20Docs/T2%202025/[GKEHub.docx](https://deakin365.sharepoint.com/:w:/r/sites/HardhatEnterprises2/Shared%20Documents/%F0%9F%95%B9%20Policy%20Deployment%20Engine/Documents/Service-Policy%20Docs/T2%202025/GKEHub.docx?d=w42cb6de7a4234be0a07619ff9a3c16a3&csf=1&web=1&e=ozqUYv)?d=w42cb6de7a4234be0a07619ff9a3c16a3&csf=1&web=1&e=ozqUYv

Pat-Stuart commented:
LGTM

I will need you to merge in the latest dev branch before merge to remove the merge conflicts. Just accept all incoming changes, this will be easy to perform in VS Code.

Sundi202 committed:
Merge origin/dev into gcp/service/GKE_Hub
